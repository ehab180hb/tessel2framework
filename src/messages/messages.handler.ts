import { fiveTypes } from '../typings/johnny-five'

export const newMessage = (lcd: fiveTypes.LCD) => (line: 0 | 1, msg: string) => {
  lcd.cursor(line, 0).print(' '.repeat(16))
  lcd.cursor(line, 0).print(msg)
}
