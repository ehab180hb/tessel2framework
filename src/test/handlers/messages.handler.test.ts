import { newMessage } from '../../messages/messages.handler'

const lcd: any = {
  cursor: (x: any, y: any) => {},
  print: (x: any) => 'x',
}

describe('Messages handler', () => {
  beforeEach(() => {
    lcd.cursor = jest.fn(() => lcd)
    lcd.print = jest.fn(() => lcd)
  })

  describe('newMessage()', () => {
    test('Clears row and prints message', () => {
      const writeMessage = newMessage(lcd)
      writeMessage(0, 'myMessage')
      expect(lcd).toMatchSnapshot()
    })
  })
})
