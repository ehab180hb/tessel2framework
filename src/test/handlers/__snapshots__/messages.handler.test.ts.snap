// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages handler newMessage() Clears row and prints message 1`] = `
Object {
  "cursor": [MockFunction] {
    "calls": Array [
      Array [
        0,
        0,
      ],
      Array [
        0,
        0,
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": [Circular],
      },
      Object {
        "isThrow": false,
        "value": [Circular],
      },
    ],
  },
  "print": [MockFunction] {
    "calls": Array [
      Array [
        "                ",
      ],
      Array [
        "myMessage",
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": [Circular],
      },
      Object {
        "isThrow": false,
        "value": [Circular],
      },
    ],
  },
}
`;
